src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -317,8 +318,9 @@ export class CodeActionModel extends Disposable {
 					// temporarilly hiding here as this is enabled/disabled behind a setting.
 					return getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
 				});
+
 				if (trigger.trigger.type === CodeActionTriggerType.Invoke) {
-					this._progressService?.showWhile(actions, 250);
+					await this._progressService?.showWhile(actions, 250);
```

Comment:
adding this await forces actions to finish before moving on (in the event of no edits, the extension comes back immediately. in the event of edits, the extension delays a little before coming back).

without waiting for actions to finish, actions is caught here after typing and quickly activating quick fixes. https://github.com/justschen/vscode/blob/14b471b2fba3fd177ad735d36ca1976154ccf026/src/vs/editor/contrib/codeAction/browser/codeActionModel.ts#L130

Reply:
we still need the await, since it's cancelled in the case where we type or make a document change. otherwise, on the call of `this.setState(new CodeActionsState.Triggered(trigger.trigger, startPosition, actions));` actions could be empty and cancelled when we type -> trigger actions

Reply:
I don't think this is the right fix. We should set the state immediately to `Triggered` here or else `actions` can be set incorrectly on multiple rapid triggers or leaked

In the case of cancellation, we still want to track that the code actions were triggered before the cancellation happens